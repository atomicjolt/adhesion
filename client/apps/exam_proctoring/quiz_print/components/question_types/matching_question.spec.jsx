import React from 'react';
import { shallow } from 'enzyme';
import Matching from './matching_question';

describe('Matching', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      answers: [
        {
          id: 1,
          text: 'Bretons'
        },
        {
          id: 2,
          text: 'Nords'
        },
        {
          id: 3,
          text: 'Argonians'
        },
      ],
      matches: [
        {
          match_id: 4,
          text: 'Black Marsh'
        },
        {
          match_id: 5,
          text: 'High Rock'
        },
        {
          match_id: 6,
          text: 'Skyrim'
        },
      ],
      question_text: '<p>Match each race to its homeland</p>',
      id: 7,
    };
    result = shallow(<Matching {...props} />);
  });

  it('renders the question text', () => {
    const element = result.find('div').at(0).props().children[0].props;
    expect(element.text).toContain('Match each race to its homeland');
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
