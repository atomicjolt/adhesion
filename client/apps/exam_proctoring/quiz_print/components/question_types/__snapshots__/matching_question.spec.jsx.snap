// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matching matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <matchingQuestion
    answers={
        Array [
            Object {
              "id": 1,
              "text": "Bretons",
            },
            Object {
              "id": 2,
              "text": "Nords",
            },
            Object {
              "id": 3,
              "text": "Argonians",
            },
          ]
    }
    id={7}
    matches={
        Array [
            Object {
              "match_id": 4,
              "text": "Black Marsh",
            },
            Object {
              "match_id": 5,
              "text": "High Rock",
            },
            Object {
              "match_id": 6,
              "text": "Skyrim",
            },
          ]
    }
    question_text="<p>Match each race to its homeland</p>"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <essayQuestion
          text="<p>Match each race to its homeland</p>"
/>,
        <div
          style={
                    Object {
                              "display": "inline-block",
                            }
          }
>
          <div
                    style={
                              Object {
                                        "border": "1px solid darkGray",
                                        "margin": "5px",
                                        "padding": "5px",
                                      }
                    }
          >
                    Bretons
          </div>
          <div
                    style={
                              Object {
                                        "border": "1px solid darkGray",
                                        "margin": "5px",
                                        "padding": "5px",
                                      }
                    }
          >
                    Nords
          </div>
          <div
                    style={
                              Object {
                                        "border": "1px solid darkGray",
                                        "margin": "5px",
                                        "padding": "5px",
                                      }
                    }
          >
                    Argonians
          </div>
</div>,
        <div
          style={
                    Object {
                              "display": "inline-block",
                              "float": "right",
                            }
          }
>
          <div
                    style={
                              Object {
                                        "border": "1px solid darkGray",
                                        "margin": "5px",
                                        "padding": "5px",
                                      }
                    }
          >
                    Black Marsh
          </div>
          <div
                    style={
                              Object {
                                        "border": "1px solid darkGray",
                                        "margin": "5px",
                                        "padding": "5px",
                                      }
                    }
          >
                    High Rock
          </div>
          <div
                    style={
                              Object {
                                        "border": "1px solid darkGray",
                                        "margin": "5px",
                                        "padding": "5px",
                                      }
                    }
          >
                    Skyrim
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "text": "<p>Match each race to its homeland</p>",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              style={
                            Object {
                                          "border": "1px solid darkGray",
                                          "margin": "5px",
                                          "padding": "5px",
                                        }
              }
>
              Bretons
</div>,
            <div
              style={
                            Object {
                                          "border": "1px solid darkGray",
                                          "margin": "5px",
                                          "padding": "5px",
                                        }
              }
>
              Nords
</div>,
            <div
              style={
                            Object {
                                          "border": "1px solid darkGray",
                                          "margin": "5px",
                                          "padding": "5px",
                                        }
              }
>
              Argonians
</div>,
          ],
          "style": Object {
            "display": "inline-block",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "question7_answer1",
            "nodeType": "host",
            "props": Object {
              "children": "Bretons",
              "style": Object {
                "border": "1px solid darkGray",
                "margin": "5px",
                "padding": "5px",
              },
            },
            "ref": null,
            "rendered": "Bretons",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "question7_answer2",
            "nodeType": "host",
            "props": Object {
              "children": "Nords",
              "style": Object {
                "border": "1px solid darkGray",
                "margin": "5px",
                "padding": "5px",
              },
            },
            "ref": null,
            "rendered": "Nords",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "question7_answer3",
            "nodeType": "host",
            "props": Object {
              "children": "Argonians",
              "style": Object {
                "border": "1px solid darkGray",
                "margin": "5px",
                "padding": "5px",
              },
            },
            "ref": null,
            "rendered": "Argonians",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              style={
                            Object {
                                          "border": "1px solid darkGray",
                                          "margin": "5px",
                                          "padding": "5px",
                                        }
              }
>
              Black Marsh
</div>,
            <div
              style={
                            Object {
                                          "border": "1px solid darkGray",
                                          "margin": "5px",
                                          "padding": "5px",
                                        }
              }
>
              High Rock
</div>,
            <div
              style={
                            Object {
                                          "border": "1px solid darkGray",
                                          "margin": "5px",
                                          "padding": "5px",
                                        }
              }
>
              Skyrim
</div>,
          ],
          "style": Object {
            "display": "inline-block",
            "float": "right",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "question7_match4",
            "nodeType": "host",
            "props": Object {
              "children": "Black Marsh",
              "style": Object {
                "border": "1px solid darkGray",
                "margin": "5px",
                "padding": "5px",
              },
            },
            "ref": null,
            "rendered": "Black Marsh",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "question7_match5",
            "nodeType": "host",
            "props": Object {
              "children": "High Rock",
              "style": Object {
                "border": "1px solid darkGray",
                "margin": "5px",
                "padding": "5px",
              },
            },
            "ref": null,
            "rendered": "High Rock",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "question7_match6",
            "nodeType": "host",
            "props": Object {
              "children": "Skyrim",
              "style": Object {
                "border": "1px solid darkGray",
                "margin": "5px",
                "padding": "5px",
              },
            },
            "ref": null,
            "rendered": "Skyrim",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <essayQuestion
            text="<p>Match each race to its homeland</p>"
/>,
          <div
            style={
                        Object {
                                    "display": "inline-block",
                                  }
            }
>
            <div
                        style={
                                    Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                        }
            >
                        Bretons
            </div>
            <div
                        style={
                                    Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                        }
            >
                        Nords
            </div>
            <div
                        style={
                                    Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                        }
            >
                        Argonians
            </div>
</div>,
          <div
            style={
                        Object {
                                    "display": "inline-block",
                                    "float": "right",
                                  }
            }
>
            <div
                        style={
                                    Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                        }
            >
                        Black Marsh
            </div>
            <div
                        style={
                                    Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                        }
            >
                        High Rock
            </div>
            <div
                        style={
                                    Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                        }
            >
                        Skyrim
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "text": "<p>Match each race to its homeland</p>",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                                Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                }
>
                Bretons
</div>,
              <div
                style={
                                Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                }
>
                Nords
</div>,
              <div
                style={
                                Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                }
>
                Argonians
</div>,
            ],
            "style": Object {
              "display": "inline-block",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "question7_answer1",
              "nodeType": "host",
              "props": Object {
                "children": "Bretons",
                "style": Object {
                  "border": "1px solid darkGray",
                  "margin": "5px",
                  "padding": "5px",
                },
              },
              "ref": null,
              "rendered": "Bretons",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "question7_answer2",
              "nodeType": "host",
              "props": Object {
                "children": "Nords",
                "style": Object {
                  "border": "1px solid darkGray",
                  "margin": "5px",
                  "padding": "5px",
                },
              },
              "ref": null,
              "rendered": "Nords",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "question7_answer3",
              "nodeType": "host",
              "props": Object {
                "children": "Argonians",
                "style": Object {
                  "border": "1px solid darkGray",
                  "margin": "5px",
                  "padding": "5px",
                },
              },
              "ref": null,
              "rendered": "Argonians",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                                Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                }
>
                Black Marsh
</div>,
              <div
                style={
                                Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                }
>
                High Rock
</div>,
              <div
                style={
                                Object {
                                                "border": "1px solid darkGray",
                                                "margin": "5px",
                                                "padding": "5px",
                                              }
                }
>
                Skyrim
</div>,
            ],
            "style": Object {
              "display": "inline-block",
              "float": "right",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "question7_match4",
              "nodeType": "host",
              "props": Object {
                "children": "Black Marsh",
                "style": Object {
                  "border": "1px solid darkGray",
                  "margin": "5px",
                  "padding": "5px",
                },
              },
              "ref": null,
              "rendered": "Black Marsh",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "question7_match5",
              "nodeType": "host",
              "props": Object {
                "children": "High Rock",
                "style": Object {
                  "border": "1px solid darkGray",
                  "margin": "5px",
                  "padding": "5px",
                },
              },
              "ref": null,
              "rendered": "High Rock",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "question7_match6",
              "nodeType": "host",
              "props": Object {
                "children": "Skyrim",
                "style": Object {
                  "border": "1px solid darkGray",
                  "margin": "5px",
                  "padding": "5px",
                },
              },
              "ref": null,
              "rendered": "Skyrim",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
